### PR TITLE
Add support for SQL databases with a PDO database class.

### PR DESCRIPTION
**Pull request by EreMaijala**
Includes database creation script for MySQL/MariaDB.

### CODE REVIEW

**Comment by LuomaJuha** [comment]
ERROR 1005 (HY000) at line 1 in file: '.../dbscripts/mysql.sql': Can't create table `recman`.`record` (errno: 150 "Foreign key constraint is incorrectly formed")
ERROR 1005 (HY000) at line 24 in file: '.../dbscripts/mysql.sql': Can't create table `recman`.`record_attrs` (errno: 150 "Foreign key constraint is incorrectly formed")
Query OK, 0 rows affected (0.033 sec)

ERROR 1005 (HY000) at line 40 in file: '.../dbscripts/mysql.sql': Can't create table `recman`.`dedup_attrs` (errno: 150 "Foreign key constraint is incorrectly formed")

Pari virhettä tuli vastaan, toimi kun ajoi toistamiseen tuon mysql.sql tiedoston

**Comment by EreMaijala** [comment]
@LuomaJuha Now fixed.

File: conf/recordmanager.ini.sample [hunk 1]
```diff
@@ -32,18 +32,28 @@ unicode_normalization_form = NFKC
...
-; Database to use
+[Database]
+; URL for a MongoDB database.
+; Use a unix socket like below, if possible. It can be an order of magnitude faster
+; than TCP/IP.
+connection = "mongodb://%2Ftmp%2Fmongodb-27017.sock"
+;connection = mongodb://{username}:{password}@localhost
+;connection = mongodb://localhost
+; MongoDB database to use
 database = recman
```
Comment (demiankatz) [inline conf/recordmanager.ini.sample:49]: If this is a Mongo-specific setting, seems like renaming it to reduce confusion might be a good idea. Or alternatively, do you have separate [Mongo] and [PDO] sections, and then pick which one is used through a  [Database] section? That might make the overall configuration less potentially confusing, especially if more backends are introduced in future.

**Review by demiankatz — COMMENTED** [review]
I finally had time to take a quick look at this. I obviously couldn't go into great depth due to lack of familiarity with the code, but I left one thought on the configuration, and I approved of what I saw in the refactoring. Do you currently use a static analysis tool of some sort? I'd strongly suggest trying PHPMD (or one of the newer ones like PHPStan or Psalm) just to make sure there are no orphaned variables/parameters after all the refactoring.

**Comment by EreMaijala** [comment]
@demiankatz Thanks for your feedback, appreciated! No static analysis tools used yet, but I'll see what I can do about that. It's a good idea! Also, splitting the database config to backend-specific sections makes sense. I'll adjust it right away.

**Comment by demiankatz** [comment]
Great! Let me know if I can do anything more to help. If you want to try PHPMD, you should be able to steal the Phing settings from VuFind pretty easily. You can also find Psalm setup in a PR: https://github.com/vufind-org/vufind/pull/1764. I haven't gotten around to investigating PHPStan yet.

File: conf/recordmanager.ini.sample (hunk 1 above)
Comment (EreMaijala) [reply conf/recordmanager.ini.sample:49]: Good idea. Now separated. Added bonus is that the MongoDB config is now fully backwards compatible without any tweaks.

Comment (demiankatz) [reply conf/recordmanager.ini.sample:49]: Great, glad I could help!

**Pull request merged**